### DISCUSSION TITLE
is key material used for AES-CCM frame security should be updated by radio all the time ?

### DISCUSSION THREAD

**Discussion by testter2** [openthread/openthread#8620] [Q&A] [answered]
Hi All,

I'm asking about the case of platform that Support OT_RADIO_TRANSMIT_SEC:
According to my understanding, In case of secured packets by Openthread like packets not under key id mode 1, then no need for radio module to update mAesKey after frame is transmitted and it updates mAesKey only if radio module processed packet security, is this correct ?

Thanks in advance.

**Comment by jwhui** [comment, marked as answer]
The radio driver should only update `mAesKey` when the MAC frame has security enabled with Key ID Mode 1.

See the following example from ot-nrf528xx:
https://github.com/openthread/ot-nrf528xx/blob/6521b600c7f916222a23b93001e553c34ba32d05/src/src/radio.c#L1253-L1256

**Reply by abtink** [reply]
One related point, it is more important (actually critical under certain configs) to properly update `mIsHeaderUpdated` and `mIsSecurityProcessed` on the `otRadioFrame` if radio platform does them (when it provides  `OT_RADIO_TRANSMIT_SEC` capability).

**Comment by testter2** [comment]
Thanks for quick reply
